### PR TITLE
Pull request for 163 - EMT GF VSC Model

### PR DESCRIPTION
**Pull request by Carlos-Alegre**
Pull request for the #163 EMT Grid Following VSC Model - colib project

### CODE REVIEW

File: pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md [hunk 1]
```diff
@@ -0,0 +1,496 @@
...
+
+<div style="background-color:rgba(0, 0, 0, 0.0470588); text-align:center; vertical-align: middle; padding:4px 0;">
+
+$$ D_1 = \frac{\sqrt{3}V_{svm} \sin(\frac{\pi}{3} - \theta_{sec1})}{2V_{DC}} $$
+$$ D_2 = \frac{V_{svm} \sin(\theta_{sec1} - \frac{\pi}{3})}{V_{DC}} $$
+</div>
+
+where $$D_1$$ and $$D_2$$ are the duty cycles of the vectors that limit the sector. The rest of the duty times will correspond to the zero-valued vectors, which will have their times distributed as $$D_{v0} = D_{v7} = \frac{1 - D_1 - D_2}{2}$$. The total time of the PWM cycle is $$T_{PWM} = \frac{1}{f_{PWM}}$$, where $$f_{PWM}$$ is the frequency of the PWM. The time of each vector is then calculated as $$T_{v} = D_{v}T_{PWM}$$.
+
+The complete block diagram of the modulation block can be seen in the following figure:
```
Comment (matoubongrain) [inline pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md:454]: maybe start by this diagram at the beginning to explain what we want, and then give the demonstration. 
what are q1, q2, q3?

File: pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md [hunk 2]
```diff
@@ -0,0 +1,496 @@
...
+
+### Limitations of current
+
+The technical constraints of the VSC can be included in the controls using saturation blocks. Depending on the desired operation mode, the converter can be set to prioritize one of the current components. These operation modes are typically defined by the grid codes, although a possible implementation could be the following:
+
+* **Normal operation**: The converter will prioritize the $$i^q$$ component, then $$i^q_{max} = I_{max}$$ and $$i^d_{max} = \sqrt{I_{max}^2 - \max{i^q, i^{q*}}} $$.
+* **Transient or fault operation**: The converter will prioritize the $$i^d$$ component, then $$i^d_{max} = I_{max}$$ and $$i^q_{max} = \sqrt{I_{max}^2 - \max{i^d, i^{d*}}^2} $$.
+
+
+### Modulation
```
Comment (matoubongrain) [inline pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md:320]: this paragraph is not clear to me...should we split and re organised.

**Review by matoubongrain — CHANGES_REQUESTED** [review]

File: pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md [hunk 3]
```diff
@@ -0,0 +1,496 @@
...
+     style="float: center; margin-right: 10px; width: 500px;" />
+</div>
+<div align = 'center'>
+Figure 1: EMT Grid Following VSC scheme <a href="#1">[1]</a>
+</div>
+<br>
+
+As it can be seen, there are several control blocks that act over the converter to determine the operating point. For a Grid Following VSC, these are:
+
+* The Clarke/Park transformation blocks, which convert the electrical variables into the *$$\alpha\beta$$* and *qd0* reference frames.
```
[the remark this answers is not in the document]
Comment (Carlos-Alegre) [reply pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md:49]: Corrected

File: pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md [hunk 4]
```diff
@@ -0,0 +1,496 @@
+---
+layout: page 
+title: EMT Grid Following Voltage Source Converter
```
[the remark this answers is not in the document]
Comment (Carlos-Alegre) [reply pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md:3]: New title: Generic EMT Grid Following Voltage Source Converter with Switching

File: pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md [hunk 5]
```diff
@@ -0,0 +1,496 @@
...
+Applying the Laplace transformation and reordering, the following relation holds:
+
+<div style="background-color:rgba(0, 0, 0, 0.0470588); text-align:center; vertical-align: middle; padding:4px 0;">
+
+$$ \frac{i^{q}(s)}{\hat{v}^{q}(s)} = \frac{1}{R + sL} $$
+$$ \frac{i^{d}(s)}{\hat{v}^{d}(s)} = \frac{1}{R + sL} $$
+
+</div>
+
+The following schematic represents the closed-loop block diagram for the current loop:
```
[the remark this answers is not in the document]
Comment (Carlos-Alegre) [reply pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md:192]: There closed-loop models the plant of the system using the new decoupling variables v-hat. This decoupling variable is calculated with the other inputs as shown in the complete block diagram, should be clearer moving this diagram to the top of the section

File: pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md [hunk 6]
```diff
@@ -0,0 +1,496 @@
...
+<div align = 'center'>
+Figure 4: Current Error Diagram
+</div>
+<br>
+
+The transfer function of this block diagram, considering $$G_c = K^{icl}_p + \frac{K^{icl}_i}{s}$$ is:
+
+<div style="background-color:rgba(0, 0, 0, 0.0470588); text-align:center; vertical-align: middle; padding:4px 0;">
+
+$$ \frac{i^{q}(s)}{i^{q*}(s)} = \frac{G_c(s)}{Ls + R + G_c(s)} $$
```
[the remark this answers is not in the document]
Comment (Carlos-Alegre) [reply pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md:208]: It is for both q and d, changing the scheme and the equations to reference both variables

File: pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md [hunk 7]
```diff
@@ -0,0 +1,496 @@
...
+
+with the PI controller
+
+<div style="background-color:rgba(0, 0, 0, 0.0470588); text-align:center; vertical-align: middle; padding:4px 0;">
+
+$$ G_{c}(s) = K_p + \frac{K_i}{s} = \frac{L}{\tau_c} + \frac{R}{\tau_c s} $$
+
+</div>
+
+The converter voltage setpoint obtained from this loop ($$v_c^{qd}$$) is then used to modulate the IGBT pulses and generate the AC voltage from the DC source. However, the modulation block can be omitted in some simplified models, and it can be considered to be directly the actual converter voltage, maintaining the notation without the setpoint indicator ($$*$$). This modulation will be explained in Section 3.6.
```
[the remark this answers is not in the document]
Comment (Carlos-Alegre) [reply pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md:239]: added

File: pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md [hunk 8]
```diff
@@ -0,0 +1,496 @@
...
+The complex power phasor, as well as the active and reactive power values, can now be obtained with:
+
+<div style="background-color:rgba(0, 0, 0, 0.0470588); text-align:center; vertical-align: middle; padding:4px 0;">
+
+$$ \underline{S} = 3 v^{qd}(i^{qd})^* = 3 (\frac{v^q - jv^d}{\sqrt{2}})(\frac{i^q + ji^d}{\sqrt{2}})$$
+$$ P = \Re(S) = \frac{3}{2} (v^q i^q + v^d i^d) $$
+$$ Q = \Im(S) = \frac{3}{2} (v^q i^d - v^d i^q) $$
+</div>
+
+Considering $$v_d = 0$$, the setpoints for current could be extracted substituting in the previous equation and using directly the grid voltage measurement $$v_q$$ obtained from the Park transformation:
```
[the remark this answers is not in the document]
Comment (Carlos-Alegre) [reply pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md:264]: added clarification

File: pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md [hunk 9]
```diff
@@ -0,0 +1,496 @@
...
+
+<div style="background-color:rgba(0, 0, 0, 0.0470588); text-align:center; vertical-align: middle; padding:4px 0;">
+$$ i^{q*} = \frac{2}{3} \frac{P^*}{v^q} $$
+$$ i^{d*} = \frac{2}{3} \frac{Q^*}{v^q} $$
+</div>
+
+However, it is not suitable to use this direct calculation under transient conditions or perturbations, and a PI controller is proposed to provide the desired response. The following block diagram shows the current setpoints output using the power setpoints:
```
[the remark this answers is not in the document]
Comment (Carlos-Alegre) [reply pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md:271]: Rephrased

File: pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md [hunk 10]
```diff
@@ -0,0 +1,496 @@
...
+<img src="{{ '/pages/models/generations/Sources/VSC/EMTGridFollowingVSC/ModulationVSC.svg' | relative_url }}"
+     alt="Modulation VSC"
+     style="float: center; margin-right: 10px; width: 100%;" />
+</div>
+<div align = 'center'>
+Figure 9: Modulation Control Blocks 
+</div>
+<br>
+
+## Open source implementations (if any)
```
[the remark this answers is not in the document]
Comment (Carlos-Alegre) [reply pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md:466]: removed

File: pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md [hunk 11]
```diff
@@ -0,0 +1,496 @@
...
+$$K^{ipl}_i = \frac{2}{3V_{peak}\tau_p}$$
+</div>
+
+The time constant $$\tau_p$$ will be larger than $$\tau_c$$, since the power loop is designed to have a slower response than the current loop.
+
+### Limitations of current
+
+The technical constraints of the VSC can be included in the controls using saturation blocks. Depending on the desired operation mode, the converter can be set to prioritize one of the current components. These operation modes are typically defined by the grid codes, although a possible implementation could be the following:
+
+* **Normal operation**: The converter will prioritize the $$i^q$$ component, then $$i^q_{max} = I_{max}$$ and $$i^d_{max} = \sqrt{I_{max}^2 - \max{i^q, i^{q*}}} $$.
```
[the remark this answers is not in the document]
Comment (Carlos-Alegre) [reply pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md:316]: i^q is the active current, and in normal operation will be set to follow the reference while the i^d will be limited by the sqrt(Imax^2 - (i^q)^2). I will rephrase it to make it clearer, and add the saturation blocks

File: pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md [hunk 12]
```diff
@@ -0,0 +1,496 @@
...
+* **Transient or fault operation**: The converter will prioritize the $$i^d$$ component, then $$i^d_{max} = I_{max}$$ and $$i^q_{max} = \sqrt{I_{max}^2 - \max{i^d, i^{d*}}^2} $$.
+
+
+### Modulation
+
+As explained in the current loop section, the converter voltage setpoint obtained is considered directly the converter voltage. However, this sinusoidal voltage is obtained from a DC source, and to generate the AC voltage, the IGBT switching is modulated using Pulse-Width Modulation (PWM). To determine the position of the IGBTs, the Space Vector PWM (SVPWM) technique is used [[9]](#9). It is based on the projection of the AC side converter voltages in a space vector. Considering the possible configurations of the IGBTs, there are a total of $$2^3 = 8$$ possible states, the resulting voltages of which can be represented in a hexagon in the $$\alpha\beta$$ reference frame:
```
[the remark this answers is not in the document]
Comment (Carlos-Alegre) [reply pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md:322]: The PWM in this case is done using this modulation method called SVPWM, instead of using carrier waves. The output of this modulation will be the On/Off position of the IGBTs in order to generate the desired voltage.
There are 8 positions, two of which are in the center of the diagram (positions 0,7), as the voltage output of this swithing state is 0.
I will rewrite the whole section to follow a better order

File: pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md [hunk 13]
```diff
@@ -0,0 +1,496 @@
...
+<img src="{{ '/pages/models/generations/Sources/VSC/EMTGridFollowingVSC/PowerLoopVSC.svg' | relative_url }}"
+     alt="Power Control Diagram"
+     style="float: center; margin-right: 10px; height: 180px;" />
+</div>
+<div align = 'center'>
+Figure 6: Power Control Diagram <a href="#3">[3]</a>
+</div>
+<br>
+
+To determine the PI controller parameters, we propose the complete power loop computing the error response, including the current loop dynamics with the obtained transfer function. The following block diagram shows the complete power loop:
```
[the remark this answers is not in the document]
Comment (Carlos-Alegre) [reply pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md:283]: We could use directly the voltage and leave it open loop, but in order to include the dynamics of the current loop the full plant is considered

File: pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md [hunk 14]
```diff
@@ -0,0 +1,496 @@
...
+$$ \omega_n = \sqrt{\frac{k_p E_m}{\tau_{PLL}}} $$
+$$ \zeta = \frac{\sqrt{\tau_{PLL} k_p E_m}}{2} $$
+</div>
+
+A typical value for the damping ratio is $$\zeta = \frac{1}{\sqrt{2}}$$, PLL natural frequency could be set to a given frequency (i.e, 50 Hz, or 1 kHz), depending on the desired response (the higher the frequency, the more aggressive). It is a choice of the user to set the desired values of the controller, setting exactly two of the four parameters ($$\omega_n$$, $$\tau_{PLL}$$, $$\zeta$$, $$k_p$$). The peak voltage $$E_m$$ is a parameter given by the utility characteristics.
+
+
+### Current control
+
+The current control is used to determine the converter voltage that has to be applied in order to maintain the current at the setpoint. The model presented uses the Internal Model Control method (IMC), described in [[7]](#7) The control is based on the electric relationship between the variables:
```
[the remark this answers is not in the document]
Comment (Carlos-Alegre) [reply pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md:153]: Added brief explanation

File: pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md [hunk 15]
```diff
@@ -0,0 +1,510 @@
...
+
+
+### Modulation
+
+As explained in the current loop section, the converter voltage setpoint obtained can be considered directly the converter voltage, in what is called an Averaged Model, since it does not consider the switching process. However, a more complete EMT simulation will model the modulation performed to obtain the sinusoidal voltage from the DC source. The modulation will determine the switching state of the 3 pairs of IGBTs in order to produce a given voltage. To do so, the technique used is the Space Vector Pulse-Width Modulation (SVPWM) [[9]](#9), which can be modelled with the following block diagram:
+
+
+<div style="background-color:rgba(0, 0, 0, 0); text-align:center; vertical-align: middle; padding:4px 0;">
+<img src="{{ '/pages/models/generations/Sources/VSC/EMTGridFollowingVSC/ModulationVSC.svg' | relative_url }}"
+     alt="Modulation VSC"
```
Comment (matoubongrain) [inline pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md:327]: just for my understanding, the input would be v_c^qd right?

File: pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md [hunk 16]
```diff
@@ -0,0 +1,510 @@
...
+<img src="{{ '/pages/models/generations/Sources/VSC/EMTGridFollowingVSC/SVPWM.svg' | relative_url }}"
+     alt="SVPWM Diagram"
+     style="float: center; margin-right: 10px; width: 500px;" />
+</div>
+<div align = 'center'>
+Figure 9: SVPWM Voltages Vector Space <a href="#9">[9]</a>
+</div>
+<br>
+
+As it can be seen, the voltage $$v^{\alpha\beta}$$ can be obtained by adding fractions of two of the vectors that delimit the region. The fraction is commonly known as duty cycle $$D_i = \frac{T_i}{T_z}$$. The duty cycle of the zero-voltage states can be calculated using the remaining time of the PWM as $$D_0 = D_7 = \frac{1 - D_i - D_j}{2}$$, while the other duty cicles have their own formulas to be calculated. Considering the symmetry of the regions, the duty cycles can be calculated referring always to the first region. Let $$D_1$$ be the duty cycle of the state with lower angle than $$v^{\alpha\beta}$$, and $$D_2$$ the duty cycle of the state with higher angle, the following calculations are made starting from the decomposition of the setpoint voltage:
```
Comment (matoubongrain) [inline pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md:348]: typo cicles

Comment (matoubongrain) [inline pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md:348]: i don't see the link between the figure and then the equations/explanations. can you put D1 on the figure?
can you put a number to the equations and the reference in the text? 
can you explain the state identifier before the table ? what is the link between the scheme and the text? can you use the same notation?

File: pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md [hunk 17]
```diff
@@ -0,0 +1,510 @@
...
+
+* **Normal operation**: The converter will follow the $$i^q$$ component setpoint, prioritizing the active power, and then $$i^d$$ will be limited by the operational limits of the converter $$i^d_{max} = \sqrt{I_{max}^2 - \max{i^q, i^{q*}}} $$.
+* **Transient or fault operation**: The converter will now prioritize the $$i^d$$ component, which will follow its reference, and $$i^q_{max} = \sqrt{I_{max}^2 - \max{i^d, i^{d*}}^2} $$.
+
+
+### Modulation
+
+As explained in the current loop section, the converter voltage setpoint obtained can be considered directly the converter voltage, in what is called an Averaged Model, since it does not consider the switching process. However, a more complete EMT simulation will model the modulation performed to obtain the sinusoidal voltage from the DC source. The modulation will determine the switching state of the 3 pairs of IGBTs in order to produce a given voltage. To do so, the technique used is the Space Vector Pulse-Width Modulation (SVPWM) [[9]](#9), which can be modelled with the following block diagram:
```
Comment (matoubongrain) [inline pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md:322]: why 3 pairs of IGBTs and not more?

File: pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md [hunk 18]
```diff
@@ -0,0 +1,510 @@
...
+     <td style="border-left:2px solid black;">$$1$$</td>
+     <td>$$1$$</td>
+     <td>$$1$$</td>
+     <td style="border-left:2px solid black;">$$0$$</td>
+     <td>$$0$$</td>
+     <td style="border-right:2px solid black;">$$0$$</td>
+</tr>
+</table>
+
+For instance, if the voltage setpoint is located in the region delimited by states $$0/7-1-6$$, the switching order would be $$000 \rightarrow 100 \rightarrow 101 \rightarrow 111 \rightarrow 101 \rightarrow 100 \rightarrow 000$$, or, if we use state identifiers, $$0 \rightarrow 1 \rightarrow 6 \rightarrow 7 \rightarrow 6 \rightarrow 1 \rightarrow 0$$. This means that the duty cycles are split in two for all the states except the 7-state. The following figure represents a switching state of the mentioned pattern:
```
Comment (matoubongrain) [inline pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md:464]: it there a function/algorithm that we can write explicitely to describe the switching sequence?

File: pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md [hunk 19]
```diff
@@ -0,0 +1,510 @@
...
+The assumptions made are:
+
+* The DC side of the converter is considered as an ideal DC voltage source $$E_{DC}$$.
+* The AC-side of the converter is connected to the grid through an AC filter formed by a resistance $$R$$ and an inductance $$L$$. The AC voltage in the converter side is $$v_{c}^{abc}$$.
+* The system is considered to be balanced, and the positive and negative sequence are not considered.
+* The synchronization with the grid is controlled by a PLL, and it is done by imposing that the $$v_d = 0$$. This design criteria allows to relate active and reactive power with one of the components of the current.
+<!-- * The switching process of the IGBTs that generate the AC wave from the modulation of the DC voltage is considered.-->
+<!-- * The transformer dynamics have been neglected. -->
+
+Although it can be used for low-frequency phenomena such as transient stability or inter-area oscillations, the number of calculations to capture necessary to capture the dynamics is much greater than Phasor models, and result in much higher execution times.
```
[the remark this answers is not in the document]
Comment (Carlos-Alegre) [reply pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md:30]: Corrected

File: pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md [hunk 20]
```diff
@@ -0,0 +1,510 @@
...
+<img src="{{ '/pages/models/generations/Sources/VSC/EMTGridFollowingVSC/PLL_ControlDiagram.svg' | relative_url }}"
+     alt="PLL Control Diagram"
+     style="float: center; margin-right: 10px; width: 700px;" />
+</div>
+<div align = 'center'>
+Figure 2: PLL Control Diagram <a href="#3">[3]</a>
+</div>
+<br>
+
+Using the Park transformation over the grid measured voltages, and assuming a small phase difference between the output of the converter and the grid voltage, the linearization $$ v^g_d = E_m \sin(\delta) \approx E_m \delta $$ can be applied, where $$E_m$$ is the peak value of the grid voltage and $$\delta = \theta - \hat{\theta}$$ is the angle difference between the measured and the reference frame angle. The linearized control structure is the following:
```
[the remark this answers is not in the document]
Comment (Carlos-Alegre) [reply pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md:114]: Added clarification that it is only done for the tunning process

File: pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md [hunk 21]
```diff
@@ -0,0 +1,510 @@
...
+<img src="{{ '/pages/models/generations/Sources/VSC/EMTGridFollowingVSC/PowerLoopVSC.svg' | relative_url }}"
+     alt="Power Control Diagram"
+     style="float: center; margin-right: 10px; height: 180px;" />
+</div>
+<div align = 'center'>
+Figure 6: Power Control Diagram <a href="#3">[3]</a>
+</div>
+<br>
+
+To determine the PI controller parameters, we include the current loop as the plant of the power loop, in order to obtain a response that accounts for the inner-loop dynamics. The following block diagram shows the complete power loop:
```
[the remark this answers is not in the document]
Comment (Carlos-Alegre) [reply pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md:283]: rephrased

File: pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md (hunk 15 above)
Comment (Carlos-Alegre) [reply pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md:327]: yes, modified the block diagram to include it

File: pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md (hunk 17 above)
Comment (Carlos-Alegre) [reply pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md:322]: Added assumption bullet point: 2-level VSC is assumed

**Review by matoubongrain — APPROVED** [review]

File: .vscode/settings.json
Comment (GJCRESYM) [inline .vscode/settings.json:1]: To be removed

File: pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md [hunk 22]
```diff
@@ -0,0 +1,520 @@
+---
+layout: page 
+title: Generic EMT Grid Following Voltage Source Converter with SVPWM Modulation
+tags: [Opensource, EMT, voltage source, converter, wind, pv, hdvc, dynawo, STEPSS]
```
Comment (GJCRESYM) [inline pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md:4]: Include double quotes to tags

File: pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md [hunk 23]
```diff
@@ -0,0 +1,520 @@
...
+
+where $$\omega$$ is the angular frequency of the grid obtained in the PLL, and $$v_{g}^d = 0$$. The block diagram associated to this system is the following:
+
+<div style="background-color:rgba(0, 0, 0, 0); text-align:center; vertical-align: middle; padding:4px 0;">
+<img src="{{ '/pages/models/generations/Sources/VSC/EMTGridFollowingVSC/CurrentControlDiagram.svg' | relative_url }}"
+     alt="Current Control Diagram"
+     style="float: center; margin-right: 10px; width: 700px;" />
+</div>
+<div align = 'center'>
+Figure 5: Current Control Diagram <a href="#3">[3]</a>
```
Comment (GJCRESYM) [inline pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md:176]: Rename to Figure 4

File: pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md [hunk 24]
```diff
@@ -0,0 +1,520 @@
...
+
+The following schematic represents the closed-loop block diagram for the current loop:
+
+<div style="background-color:rgba(0, 0, 0, 0); text-align:center; vertical-align: middle; padding:4px 0;">
+<img src="{{ '/pages/models/generations/Sources/VSC/EMTGridFollowingVSC/CurrentErrorVSC.svg' | relative_url }}"
+     alt="Current Error Diagram"
+     style="float: center; margin-right: 10px; width: 700px;" />
+</div>
+<div align = 'center'>
+Figure 4: Current Error Diagram
```
Comment (GJCRESYM) [inline pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md:211]: Rename to Figure 5

**Review by GJCRESYM — CHANGES_REQUESTED** [review]

File: .vscode/settings.json
Comment (Carlos-Alegre) [reply .vscode/settings.json:1]: removed

File: pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md (hunk 24 above)
Comment (Carlos-Alegre) [reply pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md]: numbering corrected

**Review by GJCRESYM — APPROVED** [review]
